### ISSUE TITLE
stackoverflow risk with too many attached nodes.

### ISSUE THREAD

**Issue by sofar** [minetest/minetest#3936]
This can happen with snow, plants like grass, etc. It's just a lua stack overflow, and we shouldn't recurse here at all I think.

```
2016-03-29 23:48:10: ERROR[Main]: ServerError: Lua: Runtime error from mod '*builtin*' in callback node_on_dig(): Runtime error from mod '??' in callback l_add_item(): Runtime error from mod '*builtin*' in callback luaentity_Activate(): stack overflow
2016-03-29 23:48:10: ERROR[Main]: stack traceback:
2016-03-29 23:48:10: ERROR[Main]: 	[C]: in function 'to_table'
2016-03-29 23:48:10: ERROR[Main]: 	...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:52: in function 'set_item'
2016-03-29 23:48:10: ERROR[Main]: 	...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:105: in function <...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:86>
2016-03-29 23:48:10: ERROR[Main]: 	[C]: in function 'add_entity'
2016-03-29 23:48:10: ERROR[Main]: 	...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:6: in function <...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:3>
2016-03-29 23:48:10: ERROR[Main]: 	[C]: in function 'add_item'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:123: in function 'drop_attached_node'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:174: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	...
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:204: in function 'callback'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/item.lua:511: in function </home/sofar/git/minetest/bin/../builtin/game/item.lua:439>
2016-03-29 23:48:10: ERROR[Main]: stack traceback:
2016-03-29 23:48:10: ERROR[Main]: 	[C]: in function 'add_entity'
2016-03-29 23:48:10: ERROR[Main]: 	...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:6: in function <...e/sofar/git/minetest/bin/../builtin/game/item_entity.lua:3>
2016-03-29 23:48:10: ERROR[Main]: 	[C]: in function 'add_item'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:123: in function 'drop_attached_node'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:174: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	...
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:204: in function 'callback'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/item.lua:511: in function </home/sofar/git/minetest/bin/../builtin/game/item.lua:439>
2016-03-29 23:48:10: ERROR[Main]: stack traceback:
2016-03-29 23:48:10: ERROR[Main]: 	[C]: in function 'add_item'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:123: in function 'drop_attached_node'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:174: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	...
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:175: in function 'nodeupdate_single'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:188: in function 'nodeupdate'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/falling.lua:204: in function 'callback'
2016-03-29 23:48:10: ERROR[Main]: 	/home/sofar/git/minetest/bin/../builtin/game/item.lua:511: in function </home/sofar/git/minetest/bin/../builtin/game/item.lua:439>
```

**Comment by sofar** [comment]
FYI this happened when I dug a single node in this thing:

![](http://i.imgur.com/h46sOrG.jpg)

**Comment by C1ffisme** [comment]
I can imagine that digging any part of that thing would probably cause a nuclear disaster on your computer.

But it still has to be fixed.

**Comment by sofar** [comment]
Fixed by d7908ee
